march_joint_inertia_controller/include/march_joint_inertia_controller/joint_trajectory_inertia_controller.h

```diff
@@ -0,0 +1,209 @@
+// Copyright 2020 Project March.
+
+#ifndef MARCH_HARDWARE_JOINT_TRAJECTORY_INERTIA_CONTROLLER_H
+#define MARCH_HARDWARE_JOINT_TRAJECTORY_INERTIA_CONTROLLER_H
+
+#include <cassert>
+#include <control_toolbox/pid.h>
+#include <controller_interface/controller.h>
+#include <hardware_interface/joint_command_interface.h>
+#include <hardware_interface/joint_state_interface.h>
+#include <joint_trajectory_controller/joint_trajectory_controller.h>
+#include "march_joint_inertia_controller/inertia_estimator.h"
+#include <memory>
+#include <pluginlib/class_list_macros.h>
+#include <realtime_tools/realtime_buffer.h>
+#include <realtime_tools/realtime_publisher.h>
+#include <ros/node_handle.h>
+#include <ros/time.h>
+#include <trajectory_interface/quintic_spline_segment.h>
+
+namespace joint_trajectory_controller
+{
+typedef trajectory_interface::QuinticSplineSegment<double> SegmentImpl;
+typedef JointTrajectorySegment<SegmentImpl> Segment;
+typedef typename Segment::State State;
+}  // namespace joint_trajectory_controller
+
+template <>
+class HardwareInterfaceAdapter<hardware_interface::EffortJointInterface, joint_trajectory_controller::State>
+{
+public:
+  HardwareInterfaceAdapter() : joint_handles_ptr_(nullptr)
+  {
+  }
+
+  /**
+   * \brief Initialize the controller by establishing the pointer to the joints, the pid calculators and the inertia
+   * estimators
+   */
+  bool init(std::vector<hardware_interface::JointHandle>& joint_handles, ros::NodeHandle& nh)
+  {
+    joint_handles_ptr_ = &joint_handles;
+    num_joints_ = joint_handles_ptr_->size();
+
+    // Initialize PIDs
+    pids_.resize(num_joints_);
+    for (unsigned int i = 0; i < pids_.size(); ++i)
+    {
+      ros::NodeHandle joint_nh(nh, std::string("gains/") + joint_handles[i].getName());
+
+      // Init PID gains from ROS parameter server
+      pids_[i].reset(new control_toolbox::Pid());
+      if (!pids_[i]->init(joint_nh))
+      {
+        ROS_WARN_STREAM("Failed to initialize PID gains from ROS parameter server.");
+        return false;
+      }
+    }
+
+    // Obtain inertia estimator parameters from server
+    double lambda[2];
+    int alpha_filter_size[2];
+    ros::NodeHandle rotary_estimator_nh(nh, std::string("inertia_estimator/rotary"));
+    rotary_estimator_nh.param("std_samples", samples_, 100);
+    rotary_estimator_nh.param("lambda", lambda[0], 0.96);
```

Comment:
0.96, is a bit of an arbitrary default value, may be choose one, using only the last value, seems like a safe choice.

Reply:
yeah, that's a fine suggestion. I'll change it.